### PR TITLE
fixed to be compilable on macOS 12

### LINKED ISSUE #175: v3.5.0 cannot be compiled on macos-12: `error: property 'graphicsDevices' not found on object of type 'VZVirtualMachine *'`
**Describe the bug**
v3.5.0 cannot be compiled on macos-12

**To Reproduce**
Build v3.5.0 on macos-12 on GitHub Actions

**Expected behavior**
Should be compilable

**Screenshots**
https://github.com/lima-vm/lima/actions/runs/11943778382/job/33293463704?pr=2926

```
# github.com/Code-Hex/vz/v3
virtualization_view.m:459:33: error: property 'graphicsDevices' not found on object of type 'VZVirtualMachine *'
virtualization_view.m:460:17: error: use of undeclared identifier 'VZGraphicsDevice'
virtualization_view.m:460:35: error: use of undeclared identifier 'graphicsDevice'
virtualization_view.m:460:68: error: property 'graphicsDevices' not found on object of type 'VZVirtualMachine *'
virtualization_view.m:461:21: error: use of undeclared identifier 'graphicsDevice'
virtualization_view.m:462:21: error: use of undeclared identifier 'VZGraphicsDisplay'
virtualization_view.m:462:40: error: use of undeclared identifier 'displayConfig'
virtualization_view.m:462:56: error: use of undeclared identifier 'graphicsDevice'
virtualization_view.m:463:36: error: use of undeclared identifier 'displayConfig'
```

**Environment that you use to compile (please complete the following information):**

 - Xcode version: 14.2
 - macOS Version: 12.7.6
 - mac architecture: x86_64
 - Go Version: 1.23.1

**Additional context**
NONE

### PR DESCRIPTION
**Pull request by Code-Hex**
## Which issue(s) this PR fixes:

Fixes #175

### CODE REVIEW

File: virtualization_view.m [hunk 1]
```diff
@@ -465,6 +466,7 @@ - (NSSize)getVirtualMachineSizeInPixels
             }
         });
     }
+#endif
     return sizeInPixels;
```
Comment (cfergeau) [inline virtualization_view.m:470]: Is `sizeInPixels` initialized to 0 if `INCLUDE_TARGET_OSX_14` is undefined? Or is its content undefined? I don't know objc rules for local variable initialization.

Comment (Code-Hex) [reply virtualization_view.m:470]: @cfergeau This value will become NSZeroSize. I handled this at
https://github.com/Code-Hex/vz/blob/46252d39b9cc1cfc2414239b0520437a06dbf245/virtualization_view.m#L432

**Comment by Code-Hex** [comment]
I confirmed can be compiled on macOS 12:
- 1.22: https://github.com/Code-Hex/vz/actions/runs/11951456121/job/33315129281?pr=176
- 1.23: https://github.com/Code-Hex/vz/actions/runs/11951456121/job/33315129685?pr=176

File: virtualization_view.m (hunk 1 above)
Comment (cfergeau) [reply virtualization_view.m:470]: What I suggest is `__block NSSize sizeInPixels = NSZeroSize`, example/gui-linux segfaults for me without this explicit initialization.

Comment (cfergeau) [reply virtualization_view.m:470]:
```
diff --git a/virtualization_view.h b/virtualization_view.h
index 09766a7..bdc83f5 100644
--- a/virtualization_view.h
+++ b/virtualization_view.h
@@ -6,6 +6,7 @@
 
 #pragma once
 
+#import "virtualization_helper.h"
 #import <Availability.h>
 #import <Cocoa/Cocoa.h>
 #import <Virtualization/Virtualization.h>
```
is missing for `INCLUDE_TARGET_OSX_14`

File: virtualization_view.m [hunk 2]
```diff
@@ -454,6 +454,7 @@ - (void)setupGraphicWindow
 - (NSSize)getVirtualMachineSizeInPixels
 {
     __block NSSize sizeInPixels;
```
Comment (Code-Hex) [inline virtualization_view.m:456]:
```suggestion
    __block NSSize sizeInPixels = NSZeroSize;
```

File: virtualization_view.m (hunk 1 above)
Comment (Code-Hex) [reply virtualization_view.m:470]: @cfergeau Thanks for your confirmation, I just applied it

Comment (Code-Hex) [reply virtualization_view.m:470]: Oh I didn't see this comment, I will add this

**Review by cfergeau — APPROVED** [review]

**Comment by Code-Hex** [comment]
I've confirmed to be compilable on macOS 12 in https://github.com/Code-Hex/vz-ci-test/actions/runs/11954315508/job/33324136719

**Pull request merged**